From 6aa96c6de16dfbbbff8e7e4323cdb53d3c9b6a0c Mon Sep 17 00:00:00 2001
From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Date: Sun, 2 Dec 2012 08:33:24 -0500
Subject: [PATCH 337/858] [SCSI] lpfc: fix potential NULL pointer dereference
 in lpfc_sli4_rq_put()

The dereference to 'put_index' should be moved below the NULL test.

Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Acked-by: James Smart <james.smart@emulex.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 5a25bf36b7394f11f2ff90c9e203639cd3dc041a)

Signed-off-by: Clark Williams <williams@redhat.com>
Signed-off-by: John Kacur <jkacur@redhat.com>
---
 drivers/scsi/lpfc/lpfc_sli.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/scsi/lpfc/lpfc_sli.c b/drivers/scsi/lpfc/lpfc_sli.c
index 74b67d9..d43faf3 100644
--- a/drivers/scsi/lpfc/lpfc_sli.c
+++ b/drivers/scsi/lpfc/lpfc_sli.c
@@ -438,11 +438,12 @@ lpfc_sli4_rq_put(struct lpfc_queue *hq, struct lpfc_queue *dq,
  struct lpfc_rqe *temp_hrqe;
  struct lpfc_rqe *temp_drqe;
  struct lpfc_register doorbell;
- int put_index = hq->host_index;
+ int put_index;
 
  /* sanity check on queue memory */
  if (unlikely(!hq) || unlikely(!dq))
   return -ENOMEM;
+ put_index = hq->host_index;
  temp_hrqe = hq->qe[hq->host_index].rqe;
  temp_drqe = dq->qe[dq->host_index].rqe;
 
-- 
1.8.3.1